packages/jsii-reflect/lib/assembly.ts

```diff
@@ -0,0 +1,197 @@
+import jsii = require('jsii-spec');
+import { ClassType } from './class';
+import { Dependency } from './dependency';
+import { EnumType } from './enum';
+import { InterfaceType } from './interface';
+import { Type } from './type';
+import { TypeSystem } from './type-system';
+
+export class Assembly {
+  private _typeCache?: { [fqn: string]: Type };
+  private _dependencyCache?: { [name: string]: Dependency };
+
+  constructor(
+    public readonly system: TypeSystem,
+    private readonly spec: jsii.Assembly) { }
+
+  /**
+   * The version of the spec schema
+   */
+  public get schema(): jsii.SchemaVersion {
+    return this.spec.schema;
+  }
+
+  /**
+   * The name of the assembly
+   * @minLength 1
+   */
+  public get name(): string {
+    return this.spec.name;
+  }
+
+  /**
+   * Description of the assembly, maps to "description" from package.json
+   * This is required since some package managers (like Maven) require it.
+   */
+  public get description(): string {
+    return this.spec.description;
+  }
+
+  /**
+   * The url to the project homepage. Maps to "homepage" from package.json.
+   */
+  public get homepage(): string {
+    return this.spec.homepage;
+  }
+
+  /**
+   * The module repository, maps to "repository" from package.json
+   * This is required since some package managers (like Maven) require it.
+   */
+  public get repository() {
```

Comment:
No type annotation

Reply:
It's inferred from the spec: `{ repository: string, url: string }`